src/Components/OATHeader/OATHeader.tsx

```diff
@@ -219,21 +225,66 @@ const OATHeader = ({ dispatch, state }: OATHeaderProps) => {
         }
     ];
 
-    const onFilesUpload = (files: Array<File>) => {
+    const validateSingleFileRelationships = async (
+        file,
+        newFilesErrors,
+        newFiles
+    ) => {
+        if (file) {
+            let errorFound = false;
+            // Validate contents
+            let model = await file.text();
+            model = JSON.parse(model);
+            if (model.contents) {
+                // Find within contents, if @type equals relationship, push to newFilesErrors
+                for (const content of model.contents) {
+                    if (
+                        content['@type'] &&
+                        content['@type'] === OATRelationshipHandleName
+                    ) {
+                        newFilesErrors.push(
+                            t('OATHeader.errorSingleFileHoldsRelationships', {
+                                fileName: file.name
+                            })
+                        );
+                        errorFound = true;
+                        break;
```

Comment:
I think the issue @jamiewilbraham reported is more subtle than "error on relationship found". I.e. if the graph already has the target model then this isn't an error. Same goes for untargeted relationships.

We also have to consider similar errors for components and extends.

I think the best option would be to use the DTDL validator. I.e. take the models that are currently in the graph, add the model we're importing, and run validate. This suggestion assumes that the DTDL validator checks relationships are valid (I'm not sure). Otherwise we might need to write more extensive validation code.